src/services/healthcareProfessionalService.ts

```diff
@@ -55,51 +55,67 @@ export async function addHealthcareProfessional(
     }
 }
 
-export async function addHealthcareProfessionalToFacility(facilityId: string, 
-    healthcareProfessional: gqlTypes.HealthcareProfessionalInput) {
+/**
+ * Creates a HealthcareProfessional and adds it to the listed facilities
+ * @param healthcareProfessionalInput 
+ * @returns A HealthcareProfessional object
+ */
+export async function addHealthcareProfessionalToFacility( 
+    healthcareProfessionalInput: gqlTypes.HealthcareProfessionalInput
+) {
+    const addHealthcareProfessionalResult : Result<dbSchema.HealthcareProfessional> = {
+        hasErrors: false,
+        errors: []
+    }
+
+    if (healthcareProfessionalInput.facilityIds.length == 0) {
+        throw CustomErrors.missingInput('The list of facilityIds cannot be empty.')
+    }
+
     try {
-        const facilityRef = dbInstance.collection('facilities').doc(facilityId)
         const healthcareProfessionalRef = dbInstance.collection('healthcareProfessionals').doc()
     
-        const newHealthcareProfessional = {
-            id: healthcareProfessionalRef.id, 
-            acceptedInsurance: validateInsurance(healthcareProfessional.acceptedInsurance as []),
-            degrees: mapAndValidateDegrees(healthcareProfessional.degrees as gqlTypes.Degree[]),
-            names: mapAndValidateNames(healthcareProfessional.names as gqlTypes.LocaleName[]),
-            specialties: mapAndValidateSpecialties(healthcareProfessional.specialties as gqlTypes.Specialty[]),
-            spokenLanguages: mapAndValidateLanguages(
-                healthcareProfessional.spokenLanguages as gqlTypes.SpokenLanguage[]
-            ),
-            isDeleted: false
-        }
-    
+        const newHealthcareProfessional = convertToDbHealthcareProfessional(
+            healthcareProfessionalRef.id, healthcareProfessionalInput
+        )
+
         await healthcareProfessionalRef.set(newHealthcareProfessional)
-    
-        facilityRef.update(
-            'healthcareProfessionalIds', firebase.FieldValue.arrayUnion(healthcareProfessionalRef.id)
-        ) 
-    } catch (error) {
-        throw new Error(`Error adding healthcare professional to facility: ${error}`)
-    }
-    // TODO: decide if something should be returned
-}
 
-// export async function searchHealthcareProfessionals(userSearchQuery : string[]) {
-// TODO: make it filter by params
-// const db = getFirestore()
-// const healthcareProfessionalRef = db.collection('healthcareProfessionals')
-// const snapshot = await healthcareProfessionalRef.where('id', 'in', userSearchQuery).get()
+        const facilities = healthcareProfessionalInput.facilityIds
 
-// const healthcareProfessionals = [] as HealthcareProfessional[]
+        facilities.map(async facilityId => {
```

Comment:
```suggestion
        for await (const facilityId of facilities) {
```

this `map(async` doesn't guarantee all the threads will return and could cause collisions. `for await` is the easiest way to iterate on async operations.